h2o-core/src/main/java/hex/AUC2.java

```diff
@@ -278,6 +269,50 @@ private double compute_auc() {
     return area/_p/_n;
   }
 
+  /**
+   * Compute the Area under Precision-Recall Curves using TPs and FPs.
+   * TPs and FPs are monotonically increasing.
+   * Calulation inspired by XGBoost implementation:
+   * https://github.com/dmlc/xgboost/blob/master/src/metric/rank_metric.cc#L566-L591
+   * @return
+   */
+  public double pr_auc() {
+    if (isEmpty()) {
+      return Double.NaN;
+    }
+    checkRecallValidity();
+    
+    if (_fps[_nBins-1] == 0) return 1.0;
+    if (_tps[_nBins-1] == 0) return 0.0;
+    
+    double area = 0.0;
+    assert _p > 0 && _n > 0 : "AUC-PR calculation error, sum of positives and sum of negatives should be zero.";
+    
+    double tp, prevtp = 0.0, fp, prevfp = 0.0, h, a, b;
+    for (int j = 0; j < _nBins; j++) {
+      tp = _tps[j];
+      fp = _fps[j];
+        if (tp == prevtp) {
+          a = 1.0;
+          b = 0.0;
+        } else {
+          h = (fp - prevfp) / (tp - prevtp);
+          a = 1.0 + h;
+          b = (prevfp - h * prevtp) / _p;
+        }
+        if (0.0 != b) {
+          area += (tp / _p - prevtp / _p -
```

Comment:
Tiny performance tweak: you may force to calculate `true positives / actual positives` aka `tp / _p` once. Same goes for ` prevtp / _p`

Reply:
I think this is not a crucial performance part, but I edited it as you suggested. :) Thank you!